api/overdrive.py

```diff
@@ -819,23 +819,32 @@ def circulation_lookup(self, book):
 
     def update_formats(self, licensepool):
         """Update the format information for a single book.
+
+        Incidentally updates the metadata, just in case Overdrive has
+        changed it.
         """
         info = self.metadata_lookup(licensepool.identifier)
 
         metadata = OverdriveRepresentationExtractor.book_info_to_metadata(
-            info, include_bibliographic=False, include_formats=True)
+            info, include_bibliographic=True, include_formats=True)
         if not metadata:
             # No work to be done.
             return
-        circulation_data = metadata.circulation
 
-        # The identifier in the CirculationData needs to match the
-        # identifier associated with the LicensePool -- otherwise
-        # a new LicensePool will be created.
-        circulation_data._primary_identifier.identifier = licensepool.identifier.identifier
-        replace = ReplacementPolicy(formats=True)
-        _db = Session.object_session(licensepool)
-        circulation_data.apply(_db, licensepool.collection, replace)
+        edition, ignore = self._edition(licensepool)
+
+        # The identifier in the Metadata and CirculationData need to
+        # match the identifier associated with the LicensePool --
+        # otherwise a new LicensePool will be created.
+        #
+        # TODO: It's not clear to me why this code is necessary -- why
+        # would the identifiers not match, given that we looked up
+        # this identifier? -- but removing it without understanding
+        # why it was added seems dangerous.
+        metadata.primary_identifier.identifier = licensepool.identifier.identifier
+        metadata.circulation._primary_identifier.identifier = licensepool.identifier.identifier
```

Comment:
@aslagle I've asked Edwin to review this branch as a whole but would like your opinion on this weird piece of code I noticed.

Reply:
It seems you added this code to fix tests - https://github.com/NYPL-Simplified/circulation/commit/46b171020e5bc60bce924c53bf7c5a7ac65c2c46

So if the tests are working now, probably ok to remove it?

Reply:
:blush: Thanks for doing `git blame` when I hadn't done it.

I thought that seemed like test code. Removing the code does break a test, but I'll try fixing the test. This is really ugly and should stay in the test if possible.